code/VersionedGridFieldDetailForm.php

```diff
@@ -76,8 +105,14 @@ public function canDelete() {
 		return $this->record->canDelete();
 	}
 
-	public function canPreview() {
-		return (in_array('CMSPreviewable', class_implements($this->record)) && !$this->isNew());
+	function canPreview() {
```

Comment:
We're not testing the result of the first $can assessment before launching into the second. Should the logic be something like this? Or what exactly should happen when the first assessment returns true (or false)?

if(method_exists("canPreview", $this->record)) 
    $can = $this->record->canPreview();
else
    $can = in_array('CMSPreviewable', class_implements($this->record));